idaes/generic_models/properties/core/generic/generic_property.py

```diff
@@ -2666,6 +2681,31 @@ def rule_pressure_sat_comp(b, j):
             self.del_component(self.pressure_sat_comp)
             raise
 
+    def _visc_d_phase(self):
+        try:
+            def rule_visc_d_phase(b, p):
+                return get_phase_method(b, "visc_d_phase", p)(b, p)
+            self.visc_d_phase = Expression(
+                    self.phase_list,
+                    doc="Dynamic viscosity for each phase",
+                    rule=rule_visc_d_phase)
+        except AttributeError:
+            self.del_component(self.visc_d_phase)
+            raise
+
+    def _vol_mol_phase(self):
+        try:
+            def rule_vol_mol_phase(b, p):
+                p_config = b.params.get_phase(p).config
+                return p_config.equation_of_state.vol_mol_phase(b, p)
+            self.vol_mol_phase = Expression(
```

Comment:
Just for my own reference, why declare these as expressions (i.e., `self.vol_mol_phase`, `self.visc_d_phase`, 'self.diffus_phase_comp`) as opposed to variables with constraints? I am just comparing this to how we defined properties in the NaCl and seawater property models.

Reply:
Mostly just a design decision - either approach works in the end.